lib/core/src/Ouroboros/Network/Client/Wallet.hs

```diff
@@ -225,6 +236,39 @@ chainSyncFollowTip toCardanoEra onTipUpdate =
 type RequestNextStrategy m n block
     = P.ClientPipelinedStIdle n block (Point block) (Tip block) m Void
 
+-- | Number of blocks to pipeline
+newtype Pipelining = Pipelining {getPipeliningNumber :: Natural}
+
+-- | mapping from last ingested block to the next pipeline number
+type PipeliningOfBlock block = block -> Pipelining
+
+-- | Pipelining strategies that have been in use
+
+data PipeliningStrategy block = PipeliningStrategy
+    { getPipeliningSize :: PipeliningOfBlock block 
+    , pipeliningStrategyName :: Text  
+    }
```

Comment:
```suggestion
-- | Pipelining strategies
data PipeliningStrategy block = PipeliningStrategy
    { getPipelineSize :: block -> Natural
    , pipeliningStrategyName :: Text  
    }
```

Naming things is a hard problem, but totally worth solving. 🤓 This is a good time to inline the names: `Pipelining` and `PipeliningOfBlock` are too non-descriptive to be helpful, whereas `getPipelineSize` does what it says.